src/database/operations.rs

```diff
@@ -111,6 +111,28 @@ pub(crate) async fn update_pr_merge_state(
     Ok(())
 }
 
+pub(crate) async fn update_merge_states_by_base_branch(
+    executor: impl PgExecutor<'_>,
+    repo: &GithubRepoName,
+    base_branch: &str,
+    merge_state: MergeState,
+) -> anyhow::Result<u64> {
+    let result = sqlx::query!(
```

Comment:
This query is a bit scary (although in practice pushes to the base branch should be rare, but still, there could be a lot of pull requests).

Let's add a FIXME comment here with a note to 1) Add a database index and 2) Filter PRs by state (it doesn't make sense to modify merge_state of already merged PRs), once we have PR state tracking.